centrehq/centre-tokens
# Modernize codebase

1. Auto-format code using Prettier
2. Update Truffle and Ganache to latest
3. Fix tests that broke due to Truffle Upgrade
4. Fix coverage tool
5. Use ESLint for static code analysis
6. Use latest ECMAScript syntax to clean up codebase
7. Fix lint errors, and fix tests that were really broken but were only working accidentally (e.g. due to accidental misuse of global variable assignments and test inter-dependencies)
8. Set up TypeScript (existing JS files aren't converted)
9. Set up TypeScript and Solidity linters
10. Add [tests](https://github.com/centrehq/centre-tokens/pull/296/commits/58f79be956e8bf017f37f90621b8554af7cc6c10) to check that storage slot positions stay the same between FiatTokenV1 and V2. This test also ensures Solidity upgrade does not affect storage slots going forward.
11. Update smart contract from Solidity v0.4 to Solidity v0.6, making syntactical and best practices modifications where necessary
12. Add a pre-commit hook to ensure checks are run

Note: Requires Node v12 (LTS)
